rel/filter.go

```diff
@@ -0,0 +1,328 @@
+package rel
+
+import (
+	"bytes"
+	"fmt"
+	"hash/fnv"
+
+	u "github.com/araddon/gou"
+
+	"github.com/araddon/qlbridge/expr"
+	"github.com/araddon/qlbridge/lex"
+)
+
+var (
+	_ = u.EMPTY
+
+	// Ensure Filter statement types Filter
+	_ Filter = (*FilterStatement)(nil)
+	_ Filter = (*FilterSelect)(nil)
+	// Statements with Columns
+	_ ColumnsStatement = (*FilterSelect)(nil)
+)
+
+type (
+	Filter interface {
+		String() string
+	}
+	FilterSelect struct {
+		*FilterStatement
+		Columns Columns
+	}
+	// Filter Statement is a statement of type = Filter
+	FilterStatement struct {
+		Description string       // initial pre-start comments
+		Raw         string       // full original raw statement
+		Filter      *Filters     // A top level filter
+		From        string       // From is optional
+		Limit       int          // Limit
+		HasDateMath bool         // does this have date math?
+		Alias       string       // Non-Standard sql, alias/name of sql another way of expression Prepared Statement
+		With        u.JsonHelper // Non-Standard SQL for properties/config info, similar to Cassandra with, purse json
```

Comment:
[nit] It doesn't appear we are calling parseWith() within parseFilter?    not needed for this pr, just an FYI.